src/test/packet_simulator.zig

```diff
@@ -225,136 +209,114 @@ pub fn PacketSimulator(comptime Packet: type) type {
         }
 
         /// Partitions the network. Guaranteed to isolate at least one replica.
-        fn partition_network(
-            self: *Self,
-        ) void {
+        fn auto_partition_network(self: *Self) void {
             assert(self.options.replica_count > 1);
 
-            self.is_partitioned = true;
-            self.stability = self.options.partition_stability;
-
+            var partition = self.auto_partition;
             switch (self.options.partition_mode) {
+                .none => std.mem.set(bool, partition, false),
                 .uniform_size => {
-                    // Exclude cases sz == 0 and sz == replica_count
-                    const sz =
+                    // Exclude cases partition_size == 0 and partition_size == replica_count
+                    const partition_size =
                         1 + self.prng.random().uintAtMost(u8, self.options.replica_count - 2);
-                    self.prng.random().shuffle(u8, self.replicas);
-                    for (self.replicas) |r, i| {
-                        self.partition[r] = i < sz;
+                    self.prng.random().shuffle(u8, self.auto_partition_replicas);
+                    for (self.auto_partition_replicas) |r, i| {
+                        partition[r] = i < partition_size;
                     }
                 },
                 .uniform_partition => {
                     var only_same = true;
-                    self.partition[0] =
-                        self.prng.random().uintLessThan(u8, 2) == 1;
+                    partition[0] = self.prng.random().uintLessThan(u8, 2) == 1;
 
                     var i: usize = 1;
                     while (i < self.options.replica_count) : (i += 1) {
-                        self.partition[i] =
-                            self.prng.random().uintLessThan(u8, 2) == 1;
+                        partition[i] = self.prng.random().uintLessThan(u8, 2) == 1;
                         only_same =
-                            only_same and (self.partition[i - 1] == self.partition[i]);
+                            only_same and (partition[i - 1] == partition[i]);
                     }
 
                     if (only_same) {
                         const n = self.prng.random().uintLessThan(u8, self.options.replica_count);
-                        self.partition[n] = true;
+                        self.auto_partition[n] = true;
```

Comment:
Yeah, `PacketSimulator.links` is the actual "source of truth" of which nodes are partitioned, but to keep things simple we still use the original paritioning logic to inject random partitions. Once we support asymmetric partitions it will probably be removed.